test/e2e/scrapeconfig_test.go

```diff
@@ -2119,3 +2122,204 @@ var LightSailSDTestCases = []scrapeCRDTestCase{
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           monitoringv1alpha1.OVHServiceVPS,
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Missing ApplicationKey",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           monitoringv1alpha1.OVHServiceVPS,
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid Service field",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           monitoringv1alpha1.OVHServiceVPS,
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid Service type",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           "InvalidService",
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Missing Service field",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid ConsumerKey",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           monitoringv1alpha1.OVHServiceVPS,
+				},
+			},
+		},
+		expectedError: false,
+	},
+	// {
+	// 	name: "Invalid ConsumerKey with empty value",
+	// 	scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+	// 		OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+	// 			{
+	// 				ApplicationKey:    "valid-app-key",
+	// 				ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+	// 				ConsumerKey:       v1.SecretKeySelector{Key: ""},
```

Comment:
IIUC, the `Key` string is allowed to be empty, i.e "". Try,
```suggestion
	// 				ConsumerKey:       v1.SecretKeySelector{},
```